### ISSUE TITLE
Master ticket for "quick little slimming tool"

### ISSUE THREAD

**Issue by ValWood** [pombase/website#950]
Follow on from:
https://github.com/pombase/website/issues/842

Required for release:

 - [x] MOST IMPORTANT gene name pop up needs to track down the page  for long lists

- [x]  Add the column headers (maybe at a jaunty angle ?)  (+1 from @mah11 )

- [x]  sort by name and size options. They don't really sit well with the other classifiers becasue there isn't a separate block that you can compare...
(sort by gene name is arbirtrary so it doesn't make sense to me. it just loses and the previous sorting?
Am I missing something?

- [ ]  ~I can select individual genes to appear in the list but this might also be a bit confusing, and detract from the cooler features
also I think the gene list should track what is selected by block. To do this currently, you now need to clear if you have picked up any additional genes. It was much cleaner before this feature was added.
(so essentially the row selection feature should only allow you to select one gene at a time, when you select a new gene the old row deselects)~

Desired

- [x]  Some default sorting might be optimal. At present I need to do 10 clicks to get everything visible, and presorted before I begin to sort optimally.

- [x]  Add number scale  to the "ruler" (i.e  1, 10, 20 to heavier line in the scale) 

- [x]  Include the query   (i.e decreased protein phosphorylation) More important for longer queries...
- [ ]  maybe put the query above and outside of the box?

Column reordering
- [x]   reorder inviable column to inviable, variable, viabe, no data

-  [x]  make  budding yeast absent and human absent default to the top
(mainly since they are smaller sets usually, so you see differences in a large set, can explain more easily in person)

**Comment by ValWood** [comment]
- [ ]  sort by name and size options. They don't really sit well with the other classifiers  becasue there isn't a separate block that you can compare...
(sort by gene name is arbirtrary so it doesn't make sense to me. it just loses and the previous sorting?
Am I missing something?


- [ ]  I can select individual genes to appear in the list but this might also be a bit confusing, and detract from the cooler features
also I think the gene list should track what is selected by block. To do this currently, you now need to clear if you have picked up any additional genes. It was much cleaner before this feature was added.

- [ ]   reorder inviable column to inviable, variable, viabe, no data

**Comment by mah11** [comment]
>  Probably when you 'primary sort' on a particular column, the column order should swap so that this is first

I think I would hate this. I expect the columns not to move. Downvote.

> Only include selected features in 'pop-up'

not sure I understand this one ...

>  could we track up and down the proteins on the ruler, but if you click on a column block, the block highlights with the details of the major component (I think you suggested something like this?)

How would this differ from what it's doing now?

> Add the column headers (maybe at a jaunty angle ?)

Yes. Big upvote.

**Comment by ValWood** [comment]
> Probably when you 'primary sort' on a particular column, the column order should swap so that this is first
> I think I would hate this. I expect the columns not to move. Downvote.

agreed but  I  would have x-ed this one out. I'm still going through all the outstnding options and collating

**Comment by ValWood** [comment]
> Only include selected features in 'pop-up'
> not sure I understand this one ...


I think this is an old bug. Now removed

**Comment by ValWood** [comment]
Later:

download options
https://github.com/pombase/website/issues/884

 sort colour palette
https://github.com/pombase/website/issues/883

user defined terms
https://github.com/pombase/website/issues/879

**Comment by ValWood** [comment]
~- [ ]  Make pop-up track down the page~ duplicate
~- [ ] Only include selected features in 'pop-up'~ out of date
~- [ ]  Probably when you 'primary sort' on a particular column, the column order should swap so that this is first~ out of date
~- [ ]   Possibly implement to display all options checked by default (better to see, and then uncheck)~ duplicate of below

**Comment by ValWood** [comment]
One more!

- [ ] A name and how to describe in a short phrase

Disect gene lists by multiple orthoganal features.

- [ ]  documentation

**Comment by kimrutherford** [comment]
Once the unknowns paper is out the door we should put aside some time for a chat about this.

**Comment by kimrutherford** [comment]
>  Add the column headers (maybe at a jaunty angle ?)

I've added column headers at a jaunty angle.

**Comment by kimrutherford** [comment]
>  I can select individual genes to appear in the list but this might also be a bit confusing, and detract from the cooler features also I think the gene list should track what is selected by block. To do this currently, you now need to clear if you have picked up any additional genes. It was much cleaner before this feature was added.
> (so essentially the row selection feature should only allow you to select one gene at a time, when you select a new gene the old row deselects)

I find that a bit hard to follow.  Is the summary that we should unselect everything when you click on a gene?

**Comment by kimrutherford** [comment]
>  sort by name and size options.

I've removed those.

**Comment by kimrutherford** [comment]
> make budding yeast absent and human absent default to the top

I've changed the sort order so that absent sorts to the top.

**Comment by ValWood** [comment]
>I've added column headers at a jaunty angle.

I like the column headers!  Unfortunately, the columns have disappeared...
I will check the other items when they are back.

**Comment by kimrutherford** [comment]
> Unfortunately, the columns have disappeared...

It looks fine on Firefox and Chrome for me.  What data set are you viewing?

**Comment by kimrutherford** [comment]
> Unfortunately, the columns have disappeared...

I think I have fixed that.  Perhaps you had a small dataset?  The bottom of the diagram was being cropped.  For a small dataset that could be the whole columns.  Fixed now.

**Comment by kimrutherford** [comment]
> reorder inviable column to inviable, variable, viable, no data

I've changed the order but haven't changed the labels yet.

**Comment by ValWood** [comment]
>I think I have fixed that. Perhaps you had a small dataset? The bottom of the diagram was being cropped. For a small dataset that could be the whole columns. Fixed now.

yes I had 40, fine now

**Comment by ValWood** [comment]
pre-sort sort on 
default on and 

deletion
human
budding yeast
process component function

**Comment by ValWood** [comment]
Kim suggest that sort function works by changing order of list (and that this would change order of columns.
Would always go from most to least sorted.

**Comment by kimrutherford** [comment]
> pre-sort sort on
> default on

I've made those fixes and I've changed the column order.  I've re-released pombase-dev.

> MOST IMPORTANT gene name pop up needs to track down the page for long lists

I'll look at this next but it's going to take longer.

**Comment by kimrutherford** [comment]
> Include the query (i.e decreased protein phosphorylation)

I've fixed that.

> maybe above and outside of the box?

I haven't done that as I think the whole page needs a bit of a re-design.  Let's have a chat about that sometime.

**Comment by kimrutherford** [comment]
>  Add number scale to the "ruler" (i.e 1, 10, 20 to heavier line in the scale)

I've added scale numbers.  They're at 10, 20, 30, ...  Having a number at 1 looked odd.

![gene-vis-scale-1](https://user-images.githubusercontent.com/90474/46933908-6ad68580-d0b2-11e8-91b7-6c24f51deb95.png)

**Comment by ValWood** [comment]
The tracking looks  great!
The scale bar looks great!

**Comment by ValWood** [comment]
presort good too

**Comment by ValWood** [comment]
I'm going to open a new ticket for remainder of tasks and documentation.

**Comment by ValWood** [comment]
> I'm going to open a new ticket for remainder of tasks and documentation.

OK done, not much outstanding here . Only key and documentation.
